pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -376,7 +376,9 @@ func (r *ReconcilePublishingStrategy) ensureIngressControllersExist(appIngressLi
 		var exists bool
 		for _, ingress := range ingressControllerList.Items {
 			// prevent nil pointer error
-			if ingress.Spec.Domain == "" || ingress.Status.EndpointPublishingStrategy.LoadBalancer == nil {
+			if ingress.Spec.Domain == "" ||
+				ingress.Status.EndpointPublishingStrategy == nil ||
+				ingress.Status.EndpointPublishingStrategy.LoadBalancer == nil {
```

Comment:
Would love to see this wrapped in a helper function so it reads more `if somethingThatCanBeUnderstood(...)` so folks don't need to parse the entire multi-line `if`.

Reply:
(If the functionality is abstracted like this it can be tested more pointedly.)

Reply:
this is addressed in #111